### PR TITLE
Y2K-1524 Untrusted Team Exists RPC

### PR DESCRIPTION
**Pull request by Daniel-Wang**
Purpose: We want to check if a team name is taken or not quickly for the create team wizard.

Security: 
- Only works for root teams, not leaking subteam existence
- Checks for collisions with teamnames, usernames, and reserved names

(Links with # 5257 PR on server.)

### CODE REVIEW

File: go/teams/teams.go [hunk 1]
```diff
@@ -2802,3 +2802,24 @@ func GetUntrustedTeamInfo(mctx libkb.MetaContext, name keybase1.TeamName) (info
...
+		SessionType: libkb.APISessionTypeREQUIRED,
+		Args: libkb.HTTPArgs{
+			"teamName": libkb.S{Val: name.String()},
+		},
+	})
+	if err != nil {
+		return false, err
+	}
+
+	exists, err = res.Body.AtKey("exists").GetBool()
```
Comment (zapu) [inline go/teams/teams.go]: can you look at for example:
https://github.com/keybase/client/blob/master/go/bot/service_helper.go#L25

on how to make these calls to API + extracting info easier?

File: shared/actions/typed-routes.tsx [hunk 2]
```diff
@@ -48,6 +48,7 @@ export const appendNewTeamBuilder = (teamID: TeamTypes.TeamID) =>
           filterServices: flags.teamsRedesign
             ? ['keybase', 'twitter', 'facebook', 'github', 'reddit', 'hackernews']
             : undefined,
+          goButtonLabel: 'Add',
```
Comment (zapu) [inline shared/actions/typed-routes.tsx]: unrelated?

Comment (Daniel-Wang) [reply shared/actions/typed-routes.tsx]: yeah i'm wondering how that got there...
